test/org/sagebionetworks/bridge/services/ExternalIdServiceTest.java

```diff
@@ -63,13 +71,137 @@ public void before() {
         externalIdService = new ExternalIdService();
         externalIdService.setExternalIdDao(externalIdDao);
         externalIdService.setSubstudyService(substudyService);
+        externalIdService.setAccountDao(accountDao);
     }
     
     @After
     public void after() {
         BridgeUtils.setRequestContext(null);
     }
     
+    @Test
+    public void migrateExternalIdentifier() throws Exception {
+        // setup
+        when(substudyService.getSubstudy(TestConstants.TEST_STUDY,  SUBSTUDY_ID, true))
+            .thenReturn(Substudy.create());
+        
+        ExternalIdentifier extId = ExternalIdentifier.create(TestConstants.TEST_STUDY, ID);
+        when(externalIdDao.getExternalId(TestConstants.TEST_STUDY, ID))
+            .thenReturn(extId);
+        
+        Account account = Account.create();
+        account.setId(USER_ID);
+        
+        AccountId accountId = AccountId.forExternalId(TestConstants.TEST_STUDY_IDENTIFIER, ID);
+        when(accountDao.getAccount(accountId)).thenReturn(account);
+        
+        // execute
+        externalIdService.migrateExternalIdentifier(study, ID,  SUBSTUDY_ID);
+        
+        // verify
+        assertEquals(SUBSTUDY_ID, extId.getSubstudyId());
+        assertEquals(1, account.getAccountSubstudies().size());
```

Comment:
We should also verify that we set the external ID on the account.

Reply:
Added.